code/modules/clothing/spacesuits/soviet.dm

```diff
@@ -74,22 +77,42 @@
 
 /datum/component/armor_protection_limb/proc/on_equip(obj/item/I, mob/equipper)
 	RegisterSignal(equipper, COMSIG_ATOM_BULLET_ACT, .proc/on_projectile)
+	if(has_overlay)
+		update_overlay()
 
 /datum/component/armor_protection_limb/proc/on_unequip(obj/item/I, mob/unequipper, slot)
 	if(isitem(parent))
 		if(!(I.flags_equip_slot & slotdefine2slotbit(slot)))
 			return
+		if(has_overlay)
+			usr.clear_fullscreen("helmet")
 	UnregisterSignal(unequipper, COMSIG_ATOM_BULLET_ACT)
 
 /datum/component/armor_protection_limb/proc/on_projectile(obj/item/I, var/obj/projectile/proj)
 	if(covered_limbs & GLOB.string_part_flags[proj.def_zone])
 		return adjust_health(max(0, proj.damage - round(proj.distance_travelled * proj.damage_falloff)))
 	return FALSE
 
+/datum/component/armor_protection_limb/proc/update_overlay(mob/user)
+	switch(health)
```

Comment:
```suggestion
	if(!user)
	    return
	switch(health)
```

Reply:
It only calls the proc if the parent has a user to begin with.

Reply:
If it did then it wouldn't be runtiming with `null.overlay_fullscreen()`.